Settings: re-enable Anti-spam upsell and update its copy

rcanepa opened the pull request:
#### Changes proposed in this Pull Request

* This upsell was disabled because its copy promoted Jetpack Personal which is a plan we don't offer anymore. This PR re-enables the upsell and update its copy.

#### Testing instructions

* Run this PR.
* Select a site with Jetpack Free that doesn't have Akismet or Jetpack Anti-spam.
* Visit `/settings/security/:site`.
* Verify that you see an upsell (see capture below) in the `Anti-spam` panel.
* Verify that the upsell's copy look like the one in the capture.
* Click the upsell.
* Verify that you end up in checkout with Jetpack Anti-spam in the cart.
* Do the same with a site that has either Jetpack Security or Jetpack Complete.
* Verify that you don't see the upsell.

Fixes 1196341175636977-as-1196942190681387

#### Demo
![image](https://user-images.githubusercontent.com/3418513/95103914-9aea1980-070b-11eb-8158-39ee8abf6835.png)

a8ci18n commented on client/my-sites/site-settings/spam-filtering-settings.jsx:
```diff
@@ -59,19 +58,16 @@ const SpamFilteringSettings = ( {
 			<UpsellNudge
-				description={ translate( 'Automatically remove spam from comments and contact forms.' ) }
+				description={ translate(
+					'Save time, get more responses, give your visitors a better experience - all without lifting a finger.'
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **24** times:
`translate( 'Automatically clear spam from comments and forms. Save time, get more responses, give your visitors a better experience – all without lifting a finger.' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/improve/anti-spam-upsell-copy)

a8ci18n commented on client/my-sites/site-settings/spam-filtering-settings.jsx:
```diff
@@ -59,19 +58,16 @@ const SpamFilteringSettings = ( {
-				plan={ PLAN_JETPACK_PERSONAL }
-				title={ translate( 'Defend your site against spam! Upgrade to Jetpack Personal.' ) }
+				plan={ PLAN_JETPACK_SECURITY_DAILY }
+				title={ translate( 'Automatically clear spam from comments and forms' ) }
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **21** times:
`translate( 'Automatically clear spam from comments and forms.' )` *ES Score: 11*

rcanepa commented on client/my-sites/plans-v2/utils.ts:
```diff
@@ -9,6 +9,7 @@ import React, { createElement, Fragment } from 'react';
 /**
  * Internal dependencies
  */
+import { getFeatureByKey, getFeatureCategoryByKey } from 'lib/plans/features-list';
```
Fix the circular import error (temporarily I guess?).

a8ci18n commented on client/my-sites/site-settings/spam-filtering-settings.jsx:
```diff
@@ -59,20 +59,17 @@ const SpamFilteringSettings = ( {
 		return (
 			<UpsellNudge
-				description={ translate( 'Automatically remove spam from comments and contact forms.' ) }
+				title={ translate( 'Automatically clear spam from comments and forms' ) }
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **21** times:
`translate( 'Automatically clear spam from comments and forms.' )` *ES Score: 11*

monsieur-z reviewed (changes requested):
The upsell looks great and the code as well. I noticed 2 things:
- the upsell is still shown for a site with Anti-spam
- the upsell briefly appears before disappearing for a site that includes Anti-spam (e.g. Complete). It should probably be hidden while site purchases are being fetched.

rcanepa commented:
> The upsell looks great and the code as well. I noticed 2 things:
> 
> * the upsell is still shown for a site with Anti-spam

I can't reproduce this issue. Could it be that the site you used to test this had the Akismet plugin deactivated or not installed at all? Since Jetpack Anti-spam relies on Akismet under the hood, we check for the existence of the Akismet plugin. The other check is about verifying whether the Jetpack Anti-spam feature is present in the site's plan.

> * the upsell briefly appears before disappearing for a site that includes Anti-spam (e.g. Complete). It should probably be hidden while site purchases are being fetched.

I noticed also that there is an odd state in which the `FoldableCard` appears empty because the data is still being fetched. I included in the check whether we are requesting the site's plan.

monsieur-z commented:
> I can't reproduce this issue. Could it be that the site you used to test this had the Akismet plugin deactivated or not installed at all? Since Jetpack Anti-spam relies on Akismet under the hood, we check for the existence of the Akismet plugin. The other check is about verifying whether the Jetpack Anti-spam feature is present in the site's plan.

@rcanepa, I used a brand new JN site and added Anti-spam through the store admin. 
<img width="947" alt="Screen Shot 2020-10-06 at 2 58 45 PM" src="https://user-images.githubusercontent.com/1620183/95247974-96ddfa80-07e4-11eb-840a-e57b792ec836.png">

<img width="943" alt="Screen Shot 2020-10-06 at 2 58 28 PM" src="https://user-images.githubusercontent.com/1620183/95247971-96456400-07e4-11eb-998d-8181620efea7.png">

rcanepa commented:
> @rcanepa, I used a brand new JN site and added Anti-spam through the store admin.

@monsieur-z I think I know what's going on. I don't know exactly why, but the provisioning process is not quite the same when you purchase the product through the UI as it's when you do it in Store Admin. In one case you'll end up with Akismet activated and working while in the other you won't. That's why it was showing the upsell for you.

I updated the PR to check for Jetpack Anti-spam as product. Before we were only checking for Jetpack Anti-spam and Akismet as features. Now the upsell shouldn't appear if the site has Jetpack Anti-spam regardless the state of the Akismet plugin.

rcanepa commented on client/my-sites/site-settings/spam-filtering-settings.jsx:
```diff
@@ -17,40 +17,45 @@ import FormFieldset from 'components/forms/form-fieldset';
 } ) => {
 	const { akismet: akismetActive, wordpress_api_key } = fields;
-	const isStoredKey = wordpress_api_key === currentAkismetKey;
+	const isStoredKey = wordpress_api_key === currentAkismetKey && !! wordpress_api_key;
```
Since `wordpress_api_key` could be an empty string, we don't want to set `isStoredKey` as true in this case.

rcanepa commented on client/my-sites/site-settings/spam-filtering-settings.jsx:
```diff
@@ -17,40 +17,45 @@ import FormFieldset from 'components/forms/form-fieldset';
 	const isKeyFieldEmpty = isEmpty( wordpress_api_key );
 	const isEmptyKey = isCurrentKeyEmpty || isKeyFieldEmpty;
-	const inTransition = isRequestingSettings || isSavingSettings;
+	const inTransition = isRequestingSettings || isSavingSettings || isRequestingSitePurchases;
```
Wait for the site purchases before showing anything.

rcanepa commented on client/my-sites/site-settings/spam-filtering-settings.jsx:
```diff
@@ -59,25 +64,26 @@ const SpamFilteringSettings = ( {
-		! ( hasAkismetFeature || hasAntiSpam ) &&
-		! isValidKey
-	) {
+	if ( inTransition ) {
+		return null;
+	}
```
If things are loading, don't show anything.

rcanepa commented on client/my-sites/site-settings/spam-filtering-settings.jsx:
```diff
@@ -59,25 +64,26 @@ const SpamFilteringSettings = ( {
+		return null;
+	}
+
+	if ( ! ( hasAkismetFeature || hasAntiSpamFeature || hasJetpackAntiSpamProduct ) ) {
```
Only show the upsell if the site doesn't have any of the following: Akismet as a feature, Jetpack Anti-spam as feature, or Jetpack Anti-spam as a product.

monsieur-z reviewed (approved):
LGTM! I don't know if that's something we still want to do, but with the old flow it should probably redirect to the Personal (or Business?) plan.

rcanepa commented:
> LGTM! I don't know if that's something we still want to do, but with the old flow it should probably redirect to the Personal (or Business?) plan.

Good question. Since the new approach doesn't explicitly mention any plan and it takes users directly to the checkout, it would work also in the old flow. For that reason, I'll leave as it is.

rcanepa commented:
Rebased to solve conflicts.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/5029139

Thank you @rcanepa for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/5029139).
